src/layer.rs

```diff
@@ -480,6 +454,20 @@ impl Default for Layer {
     }
 }
 
+/// [UFO layer info], corresponding to a 'layerinfo.plist' file in a layer directory.
+///
+/// This will be merged into a [`Layer`].
+///
+/// [UFO layer info]: https://unifiedfontobject.org/versions/ufo3/glyphs/layerinfo.plist/
+#[derive(Debug, Clone, Deserialize)]
```

Comment:
Just a thought: given that this is just a helper type used during deserialization, I would probably put this all in a little helper fn like:


```rust
fn parse_layer_info(path: &Path) -> Result<(Option<Color>, Plist), SomeError> {
    #[derive(Deserialize)]
    struct LayerInfoHelper {
        color: Option<Color>,
        #[serde(default)]
        lib: Plist,
    }
    // do your logic here, and return the error or the contents
}
```

The idea is to make it clear that the type is a helper by keeping it out of the main scope, and having all of the code relevant for deserializing this type be defined in the same place; you can just look at this function and know exactly what's happening, without needing to jump around.

Reply:
Good thinking. Maybe one day we don't have to do gymnastics to support different UFO normalization methods, then we can break it out again.